napari/_qt/widgets/qt_arbitary_int_spinbox.py

```diff
@@ -0,0 +1,71 @@
+import logging
+
+import numpy as np
+from qtpy.QtCore import Signal
+from qtpy.QtWidgets import QDoubleSpinBox
+
+logger = logging.getLogger(__name__)
+
+
+class ArbitraryIntDoubleSpinBox(QDoubleSpinBox):
```

Comment:
Can you add a doc string explaining when you'd want to use this `ArbitraryIntDoubleSpinBox` compared to other SpinBoxes offered by Qt (`QSpinBox` or `QDoubleSpinBox`).